73.md

```diff
@@ -0,0 +1,190 @@
+    // ... other tags
+  ], author.privateKey, author.pubkey),
+  // ...
+}
+``` 
+
+### Viewing permissions
+
+Viewing permissions are shared by creating a new Nostr Private key, passing it to each user via a `p` tag, and encrypting the `.content` to that key. 
+
+The viewing private key is NIP-44-encrypted to each `p` tag and placed as a 4th value in each tag.
+
+The `.content` is then encrypted by a conversation key between the new author's private key and the viewing's public key. 
+
+```js
+val viewingKeyPair = nostr.generateKeyPair()
+
+{
+  "pubkey": author.pubkey
+  "kind": 35337,
+  "tags": [
+    ["d", "<unique identifier>"]
+    ["p", "<pubkey 1>", "<relay url>", nip44Encrypt(viewingKeyPair.privateKeyHex, author.privateKey, "<pubkey 1>") ]
+    ["p", "<pubkey 2>", "<relay url>", nip44Encrypt(viewingKeyPair.privateKeyHex, author.privateKey, "<pubkey 2>") ]
+    ["p", "<pubkey 3>", "<relay url>", nip44Encrypt(viewingKeyPair.privateKeyHex, author.privateKey, "<pubkey 3>") ]
+  ],
+  "content": nip44Encrypt([
+    ["title", "Name of this spreadsheet"], // private title
+    ["data", "<sheet name>", "<column letter>", "<row number>", "<value>"], // private data
+    ["data", "<sheet name>", "<column letter>", "<row number>", "<value>", "<style>"],
+    ["style", "<TBD>"], // Need to specify all options here. 
+    // ... other tags
+  ], author.privateKey, viewingKeyPair.publicKey),
+  "sig": signWith(author.privateKey)
+  // ...
+}
+``` 
+
+with `nip44Encrypt(textToEncrypt, sender.privatekey, receiver.publickey)`.
+
+To decrypt, receivers MUST: 
+1. find the ciphertext in the `p`-tag for their key
+2. decrypt the ciphertext with `nip44Decrypt(tag[3], user.privatekey, event.pubkey)` to get the viewing private key
+3. use the viewing private key to decrypt the `.content` with `nip44Decrypt(event.content, viewing.privatekey, event.pubkey)`
+
+Clients SHOULD include the author as a `p` tag to make sure the author can retieve the viewing key on updates of this event. 
+
+### Editing permissions
+
+To share editing permissions, a new shareable key is needed to encrypt and sign the replaceable event. The editting key MUST be shared with other users through their `p` tags. At the same time, the viewing key can be shared with users to decrypt the event, but it doesn't allow them to change it. 
+
+```js
+val edittingKeyPair = nostr.generateKeyPair()
+val viewingKeyPair = nostr.generateKeyPair()
+
+{
+  "pubkey": edittingKeyPair.publicKey
+  "kind": 35337,
+  "tags": [
+    ["d", "<unique identifier>"]
+    ["p", "<pubkey 1>", "<relay url>", nip44Encrypt(edittingKeyPair.privateKeyHex, edittingKeyPair.privateKey, "<pubkey 1>") ]
+    ["p", "<pubkey 2>", "<relay url>", nip44Encrypt(edittingKeyPair.privateKeyHex, edittingKeyPair.privateKey, "<pubkey 2>") ]
+    ["p", "<pubkey 3>", "<relay url>", nip44Encrypt(viewingKeyPair.privateKeyHex,  edittingKeyPair.privateKey, "<pubkey 3>") ] // view only
```

Comment:
I see 2 problems here:
1) Editor doesn't know if there is a viewing key, so attempting to decrypt content will fail if there is a viewer
2) Editor doesn't know the viewing key when there is one, so editor can't add new viewers nor decrypt content

Although cumbersome, editor does could decrypt all `p` tags till reaching the third one and finding the viewingKeyPair.privateKeyHex. Though if an editor removes all viewers without re-encrypting the content, the content may be lost (cause the viewing key would be gone) so we also need an easy way to know how many viewers exist so to know when to re-encrypt the content.

One way of dealing with all of this is adding "Editor" and "Viewer" markers to the p tag. Here markers make no harm afaik.

Reply:
> Editor doesn't know if there is a viewing key

They have to decrypt all p-tags to see if there is a viewing key. 

> Editor doesn't know the viewing key when there is one

Editors can decrypt the viewing key. 

>  "Editor" and "Viewer" markers

There is no way to enforce that these markers actually represent what they are supposed to. So, it's better to just decrypt all tags.